.github/workflows/buildandtest.yml

```diff
@@ -0,0 +1,52 @@
+name: BuildAndTest
+
+on:
+  push:
+    branches: [ development ]
+  pull_request:
+    branches: [ development ]
+
+env:
+  # Customize the CMake build type here (Release, Debug, RelWithDebInfo, etc.)
+  BUILD_TYPE: Release
+
+jobs:
+  build:
+    name: Build and test on ${{matrix.os}} with threading ${{matrix.threading}} and sharedlib $${{matrix.sharedlib}} 
+    # The CMake configure and build commands are platform agnostic and should work equally
+    # well on Windows or Mac.  You can convert this to a matrix build if you need
+    # cross-platform coverage.
+    # See: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/managing-complex-workflows#using-a-build-matrix
+    runs-on: ${{matrix.os}}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-18.04, windows-latest, macOS-latest]
+        sharedlib: [ON, OFF]
+        threading: [ON, OFF]
+        
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+         submodules: true
+
+    - name: Install valgrind
+      run: sudo apt-get install -y valgrind
+      if: ${{matrix.os == 'ubuntu-18.04'}}
+
+    - name: Configure CMake
+      # Configure CMake in a 'build' subdirectory. `CMAKE_BUILD_TYPE` is only required if you are using a single-configuration generator such as make.
+      # See https://cmake.org/cmake/help/latest/variable/CMAKE_BUILD_TYPE.html?highlight=cmake_build_type
+      run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} -DUS_BUILD_TESTING:BOOL=ON -DBUILD_SHARED_LIBS:BOOL=$${{matrix.sharedlib}} -DUS_BUILD_EXAMPLES:BOOL=ON -DUS_ENABLE_THREADING_SUPPORT:BOOL=${{matrix.threading}}
```

Comment:
Why are there two "$" symbols in front of "{{matrix.sharedlib}}"? Earlier in the line, there is only one "$" in front of "{{env.BUILD_TYPE}}".

Reply:
$$ was a typo. it should be $